src/vm/eetoprofinterfaceimpl.cpp

```diff
@@ -3184,6 +3230,78 @@ HRESULT EEToProfInterfaceImpl::JITCompilationStarted(FunctionID functionId,
     }
 }
 
+HRESULT EEToProfInterfaceImpl::DynamicMethodJITCompilationFinished(FunctionID functionId)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        CAN_TAKE_LOCK;
+
+        // The JIT / MethodDesc code likely hold locks while this callback is made
+
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+    
+    CLR_TO_PROFILER_ENTRYPOINT((LF_CORPROF, 
+                                LL_INFO1000, 
+                                "**PROF: DynamicMethodJITCompilationFinished 0x%p.\n", 
+                                functionId));
+
+    _ASSERTE(functionId);
+
+    // Should only be called on profilers that support ICorProfilerCallback8
```

Comment:
Its not clear if this comment is suggesting this current method shouldn't have been called, or that this method shouldn't make the call to m_pCallback8->DynamicMethodJITCompilationFinished. I think it is clearer to eliminate the comment.